core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -124,7 +124,17 @@ public void heartbeat() {
     Command cmdFromMaster = null;
     List<alluxio.grpc.Metric> metrics = new ArrayList<>();
     for (Metric metric : MetricsSystem.allWorkerMetrics()) {
-      metrics.add(metric.toProto());
+      try {
+        metrics.add(metric.toProto());
```

Comment:
why do we hit NPE?

Reply:
This is because previously we change PROCESS_TYPE=MASTER in the tests which cause the metrics to be messy. Some metrics are master metrics and  `mHostname=null` which make the `Metric.toProto()` to throw NPE.

When `PROCESS_TYPE=CLIENT`, all metrics have `mHostname` and will not cause NPE now.